repo.go

```diff
@@ -507,36 +509,53 @@ func (r *Repo) setTopLevelMeta(roleFilename string, meta interface{}) error {
 	return r.local.SetMeta(roleFilename, b)
 }
 
-func (r *Repo) Sign(roleFilename string) error {
-	role := strings.TrimSuffix(roleFilename, ".json")
+// SignPayload signs the given payload using the key(s) associated with role.
+//
+// It returns the total number of keys used for signing, 0 (along with
+// ErrInsufficientKeys) if no keys were found, or -1 (along with an error) in
+// error cases.
+func (r *Repo) SignPayload(role string, payload *data.Signed) (int, error) {
 	if !roles.IsTopLevelRole(role) {
-		return ErrInvalidRole{role, "only signing top-level metadata supported"}
+		return -1, ErrInvalidRole{role, "only signing top-level metadata supported"}
 	}
 
-	s, err := r.SignedMeta(roleFilename)
+	keys, err := r.getSortedSigningKeys(role)
 	if err != nil {
-		return err
+		return -1, err
+	}
+	if len(keys) == 0 {
+		return 0, ErrInsufficientKeys{role}
```

Comment:
Nit: I know this was already the case before this PR, but isn't it weird that the error is named "insufficient keys" when it is only returned when *no keys* where found? Shouldn't it be called `ErrNoKeys`? I would expect an "insufficient keys" errors to be returned when the number of keys we found is lesser than the threshold for this role.

Reply:
Renamed `ErrInsufficientKeys` to `ErrNoKeys`